### PR TITLE
use selected subscriptions

### PR DESCRIPTION
**Pull request by ranasaria**
This change exposes selectedSubscription api from azurecore extension and uses that api to populate the subscriptions dropdown in wizards.

This PR fixes first part of #12261

### CODE REVIEW

File: extensions/azurecore/src/azureResource/utils.ts [hunk 1]
```diff
@@ -258,3 +258,30 @@ export async function getSubscriptions(appContext: AppContext, account?: azdata.
 	}));
 	return result;
 }
+
+export async function getSelectedSubscriptions(appContext: AppContext, account?: azdata.Account, ignoreErrors: boolean = false): Promise<GetSubscriptionsResult> {
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/azureResource/utils.ts:262]: I would rather this is just a parameter on getSubscriptions, filteredOnly or something. No reason to duplicate all this logic.

File: extensions/resource-deployment/src/localizedConstants.ts [hunk 2]
```diff
@@ -3,14 +3,16 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+//import { EOL } from 'os';
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/localizedConstants.ts:6]: remove

File: extensions/resource-deployment/src/localizedConstants.ts [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
 import * as nls from 'vscode-nls';
 import { OptionsSourceType } from './helpers/optionSources';
 import { FieldType, OptionsType } from './interfaces';
 
 const localize = nls.loadMessageBundle();
 
 export const account = localize('azure.account', "Azure Account");
-export const subscription = localize('azure.account.subscription', "Subscription");
+export const subscription = localize('azure.account.subscription', "Subscription (selected subset)");
+export const subscriptionDescription = localize('azure.account.subscriptionDescription', "Only subscriptions selected in Azure navigation tree are shown. Change the set of selected subscriptions in 'CONNECTIONS->AZURE' navigation tree to see others)");
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/localizedConstants.ts:15]: Don't use capitals like this. My suggestion would be `Change the currently selected subscriptions through the 'Select Subscriptions' action on an account listed in the 'Azure' tree view of the 'Connections' viewlet.

Although I'm wondering if maybe we should also have a checkbox for displaying all subs or just filtered ones. Not sure on that though so let's go with this for now and if we get negative feedback we can look at it again then.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: extensions/azurecore/src/azureResource/utils.ts (hunk 1 above)
Comment (ranasaria) [reply extensions/azurecore/src/azureResource/utils.ts:262]: making it a parameter in the api, doing it here would be too convulated as selectedSubscriptions do not need token logic at all.

File: extensions/resource-deployment/src/localizedConstants.ts (hunk 3 above)
Comment (ranasaria) [reply extensions/resource-deployment/src/localizedConstants.ts]: later we can actually sync filtering done in deployment back to the azure viewlet as well. That is the experience that the azure portal has.

**Pull request merged**